released/usecases/services_apps/sap_build_code/step1/main.tf

```diff
@@ -0,0 +1,328 @@
+# ------------------------------------------------------------------------------------------------------
+# SUBACCOUNT SETUP
+# ------------------------------------------------------------------------------------------------------
+# Setup subaccount domain (to ensure uniqueness in BTP global account)
+resource "random_id" "subaccount_domain_suffix" {
+  byte_length = 12
+}
+# ------------------------------------------------------------------------------------------------------
+# Creation of subaccount
+# ------------------------------------------------------------------------------------------------------
+resource "btp_subaccount" "build_code" {
+  name      = var.subaccount_name
+  subdomain = join("-", ["sap-build-code", random_id.subaccount_domain_suffix.hex])
+  region    = lower(var.region)
+}
+
+# ------------------------------------------------------------------------------------------------------
+# CLOUDFOUNDRY PREPARATION
+# ------------------------------------------------------------------------------------------------------
+#
+# Fetch all available environments for the subaccount
+data "btp_subaccount_environments" "all" {
+  subaccount_id = btp_subaccount.build_code.id
+}
+# ------------------------------------------------------------------------------------------------------
+# Take the landscape label from the first CF environment if no environment label is provided
+# (this replaces the previous null_resource)
```

Comment:
I would omit this as a user does not know about any previous version

```suggestion

```

Reply:
Makes sense. Thanks.